### PR TITLE
correctly set `previous` dist tag on patches

### PR DESCRIPTION
**Pull request by aruniverse**
if the pkg@localVer doesnt exist yet, then it hasnt been published (obviously), so determine the dist tag

### CODE REVIEW

**Comment by aruniverse** [comment]
@mergifyio backport release/4.0.x

**Comment by aruniverse** [comment]
@mergifyio backport release/3.7.x

File: common/scripts/gather_packages.py [hunk 1]
```diff
@@ -121,27 +121,30 @@ def determineDistTag(branchName, currentVer, latestVer, previousVer):
...
     print ("The package does not yet exist.  Copying " + packageName + " to staging area.")
     shutil.copy(artifact, stagingDir)
-    continue
 
   if 0 != len(serverVer):
     print ("The version already exists.  Skipping...")
     continue
 
   packagesToPublish = True
   print ("Local version is newer than on the server.  Copying package " + packageName + " to staging area.")
```
Comment (nick4598) [inline common/scripts/gather_packages.py:130]: Can we pick one of the two messages? We'll print this one and this print ("The package does not yet exist.  Copying " + packageName + " to staging area.") now, right?

File: common/scripts/gather_packages.py [hunk 2]
```diff
@@ -121,27 +121,30 @@ def determineDistTag(branchName, currentVer, latestVer, previousVer):
...
+  if not os.path.exists(os.path.join(stagingDir, baseName)):
+    shutil.copy(artifact, stagingDir)
 
-  command = "npm view " + packageName + " dist-tags.previous"
+  command = "npm dist-tag ls " + packageName
   proc = subprocess.Popen(command, stdin = subprocess.PIPE, stdout = subprocess.PIPE, shell=True)
-  previousVer = proc.communicate()[0]
-  if len(previousVer) == 0:
-    print("No version found for dist-tag 'previous'")
+  distTags = proc.communicate()[0]
```
Comment (nick4598) [inline common/scripts/gather_packages.py:136]: We're also figuring out distTags for each package, but we only need to set these values once right?

Should we just do the first package we find that is copied to the staging area? Not sure how much it impacts the runtime

Comment (Jake-Screen) [reply common/scripts/gather_packages.py:136]: added check to see if versions were already set

File: common/scripts/gather_packages.py (hunk 1 above)
Comment (Jake-Screen) [reply common/scripts/gather_packages.py:130]: chose to use the "local version is newer" message

File: common/scripts/gather_packages.py (hunk 2 above)
Comment (Jake-Screen) [reply common/scripts/gather_packages.py]: now i have

**Review by nick4598 — APPROVED** [review]

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**